src/test/java/org/jenkinsci/plugins/workflow/support/pickles/ExecutorPickleTest.java

```diff
@@ -71,4 +78,33 @@
         });
     }
 
+    @Issue("JENKINS-42556")
+    @Test public void anonDiscover() {
+        r.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                r.j.jenkins.setSecurityRealm(r.j.createDummySecurityRealm());
+                r.j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy().
+                    grant(Jenkins.ADMINISTER).everywhere().to("admin").
+                    grant(Jenkins.READ, Item.DISCOVER).everywhere().toEveryone());
+                r.j.jenkins.save(); // TODO pending https://github.com/jenkinsci/jenkins/pull/2790
+                DumbSlave remote = r.j.createSlave("remote", null, null);
+                WorkflowJob p = r.j.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition("node('remote') {semaphore 'wait'}", true));
+                SemaphoreStep.waitForStart("wait/1", p.scheduleBuild2(0).waitForStart());
+                remote.toComputer().setTemporarilyOffline(true, new OfflineCause.UserCause(User.get("admin"), "hold"));
```

Comment:
Otherwise `ExecutorPickle` could resume too quickly without ever even trying to print a message about its progress.